src/Database/Traits/Revisionable.php

```diff
@@ -157,7 +157,8 @@ protected function revisionableCleanUp()
 
     protected function revisionableGetCastType($attribute)
     {
-        if (in_array($attribute, $this->getDates())) {
+        // dates property deprecated in Laravel 10, no longer used in getDates()
+        if (in_array($attribute, $this->getDates()) || $this->isDateCastable($attribute) || in_array($attribute, $this->dates)) {
```

Comment:
Is this required if we're overriding the getDates() method?

Reply:
I frankly don't know.

Reply:
I checked Laravel's code and I would say yes, this method could be removed.

Reply:
Scratch that, the `cast` db column for the system_revisions is needed and is seperate from the main model's cast property... So I think this is still needed.

Reply:
I was referring to the fact that the change here has an associated comment of `dates property deprecated in Laravel 10, no longer used in getDates()` but we override the `getDates()` method to undeprecate the `dates` property so I'm not sure if this change is still required here.

Reply:
I understand, but this might introduce a different behavior in the Revisionable trait if we remove this code. Your call.